Foundation/JSONSerialization.swift

```diff
@@ -577,9 +577,12 @@ private struct JSONWriter {
             writer(" ")
         }
     }
-    
+
     //[SR-2151] https://bugs.swift.org/browse/SR-2151
     private mutating func _serializationString(for number: NSNumber) -> String {
+        if _CFNumberGetType2(number._cfObject) == kCFNumberSInt128Type {
```

Comment:
No - it's the `_CFNumberGetType2` that is the problem. `_cfObject` is just our way to get toll-free bridging to work here.